internal/process/queue.go

```diff
@@ -15,78 +17,111 @@ type Executor interface {
+		healthCheckFrequencyTime: healthCheckFrequencyTime,
 	}
 }
 
 func (q *Queue) Add(processId string) {
 	q.queue.Add(processId)
+	q.log.Infof("added item %s to the queue %s, queue length is %d", processId, q.name, q.queue.Len())
 }
 
 func (q *Queue) AddAfter(processId string, duration time.Duration) {
 	q.queue.AddAfter(processId, duration)
+	q.log.Infof("item %s will be added to the queue %s after duration of %d, queue length is %d", processId, q.name, duration, q.queue.Len())
 }
 
 func (q *Queue) ShutDown() {
+	q.log.Infof("shutting down the queue, queue length is %d", q.queue.Len())
 	q.queue.ShutDown()
 }
 
 func (q *Queue) Run(stop <-chan struct{}, workersAmount int) {
+	q.log.Infof("starting workers, queue length is %d", q.queue.Len())
 	for i := 0; i < workersAmount; i++ {
 		q.waitGroup.Add(1)
-		q.createWorker(q.queue, q.executor.Execute, stop, &q.waitGroup, q.log)
+
+		workerLogger := q.log.WithField("workerId", i)
+		workerLogger.Infof("starting worker with id %d", i)
+
+		q.createWorker(q.queue, q.executor.Execute, stop, &q.waitGroup, workerLogger, fmt.Sprintf("%s-%d", q.name, i))
 	}
+
+	// go routine for checking worker execution times and warning if worker has not run for specified amount of time
+	go func() {
+		wait.Until(func() {
+			q.HealthCheck()
+		}, q.healthCheckFrequencyTime, stop)
+	}()
+
 }
 
 // SpeedUp changes speedFactor parameter to reduce time between processing operations.
 // This method should only be used for testing purposes
 func (q *Queue) SpeedUp(speedFactor int64) {
 	q.speedFactor = speedFactor
+	q.log.Infof("queue speed factor set to %d", speedFactor)
 }
 
-func (q *Queue) createWorker(queue workqueue.RateLimitingInterface, process func(id string) (time.Duration, error), stopCh <-chan struct{}, waitGroup *sync.WaitGroup, log logrus.FieldLogger) {
+func (q *Queue) createWorker(queue workqueue.RateLimitingInterface, process func(id string) (time.Duration, error), stopCh <-chan struct{}, waitGroup *sync.WaitGroup, log logrus.FieldLogger, nameId string) {
 	go func() {
-		wait.Until(q.worker(queue, process, log), time.Second, stopCh)
+		log.Info("worker routine - starting")
+		wait.Until(q.worker(queue, process, log, nameId), time.Second, stopCh)
 		waitGroup.Done()
+		log.Info("worker done")
 	}()
 }
 
-func (q *Queue) worker(queue workqueue.RateLimitingInterface, process func(key string) (time.Duration, error), log logrus.FieldLogger) func() {
+func (q *Queue) worker(queue workqueue.RateLimitingInterface, process func(key string) (time.Duration, error), log logrus.FieldLogger, nameId string) func() {
 	return func() {
 		exit := false
 		for !exit {
 			exit = func() bool {
 				key, shutdown := queue.Get()
 				if shutdown {
+					log.Infof("shutting down")
 					return true
 				}
+
 				id := key.(string)
 				log = log.WithField("operationID", id)
+				log.Infof("about to process item %s, queue length is %d", id, q.queue.Len())
+				q.logAndUpdateWorkerTimes(key.(string), nameId, log)
```

Comment:
Since `Get` blocks when queue is empty we will get false warnings/alerts .

Reply:
Right, I will change "Warn" to "Info"

Reply:
I do not get the point of this change.

Reply:
What I mean is that this is not a health check. Or oversensitive health check.

Reply:
We can change the name if you want. Do you have a proposal?

Reply:
`logWorkerExecutionTimes`